File: fact.java

package codechef;
import java.math.BigInteger;
import java.util.*;
public class fact {
    public static void main(String[] args)throws java.lang.Exception
    {
        Scanner sc=new Scanner(System.in);
        int n,t;
        t=sc.nextInt();
        for(int i=0;i<t;i++){
          n=sc.nextInt();
            BigInteger factorial=BigInteger.ONE;
          for(int j=2;j<=n;j++)
          {
              factorial=factorial.multiply(new BigInteger(String.valueOf(j)));
          }
          System.out.println(factorial);
        } 
       
        
        
    }
}

File: fact.cpp
#include<iostream>
#include<biginteger.h>
using namespace std;
int main()
{
     int t,n;
     big_int fact;
cin>>t;
for(int i=0;i<t;i++)
{fact=1;
    cin>>n;
    for(int j=n;j>=1;j--)
    {
        fact=fact*j;
    }
    cout<<fact<<"\n";
}
return 0;
}
